src/components/inputs/rich-text-input/editor.js

```diff
@@ -0,0 +1,347 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import { useIntl } from 'react-intl';
+import Button from './button';
+import StyleDropdown from './style-dropdown';
+import MarkButton from './buttons/mark-button';
+import Tooltip from '../../tooltip';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import CollapsibleMotion from '../../collapsible-motion';
+import Spacings from '../../spacings';
+import {
+  BoldIcon,
+  ItalicIcon,
+  UnorderedListIcon,
+  OrderedListIcon,
+  UnderlineIcon,
+  RedoIcon,
+  UndoIcon,
+} from './icons';
+import { AngleUpIcon, AngleDownIcon } from '../../icons';
+import Constraints from '../../constraints';
+import FlatButton from '../../buttons/flat-button';
+import {
+  Toolbar,
+  ToolbarMainControls,
+  ToolbarRightControls,
+  EditorContainer,
+  Container,
+} from './rich-text-input.styles';
+import Divider from './divider';
+import MultiDropdown from './multi-dropdown';
+import messages from './messages';
+
+const tooltipStyles = {
+  body: {
+    zIndex: 9999,
+  },
+};
+const DEFAULT_NODE = 'paragraph';
+
+const NUMBERED_LIST = 'numbered-list';
+const BULLETED_LIST = 'bulleted-list';
+const LIST_ITEM = 'list-item';
+
+const COLLAPSED_HEIGHT = 64;
+
+const dropdownOptions = [
+  { label: 'Strikethrough', value: 'strikethrough' },
+  { label: 'Superscript', value: 'superscript' },
+  { label: 'Subscript', value: 'subscript' },
+];
+
+const Editor = props => {
+  const intl = useIntl();
+  const { editor } = props;
+  const ref = React.useRef();
+  const [renderToggleButton, setRenderToggleButton] = React.useState(false);
+
+  React.useEffect(() => {
+    if (ref.current.clientHeight > COLLAPSED_HEIGHT && !renderToggleButton) {
+      setRenderToggleButton(true);
+    }
+    if (ref.current.clientHeight <= COLLAPSED_HEIGHT && renderToggleButton) {
+      setRenderToggleButton(false);
+    }
+  }, [editor.value, renderToggleButton]);
+
+  const activeBlock =
+    (editor.value.blocks.first() && editor.value.blocks.first().type) || '';
+
+  const activeMarks = Array.from(editor.value.activeMarks).map(
+    mark => mark.type
+  );
+
+  const activeMoreStyleMarks = activeMarks.filter(activeMark =>
+    dropdownOptions.some(dropdownOption => activeMark === dropdownOption.value)
+  );
+
+  const hasBlock = type => editor.value.blocks.some(node => node.type === type);
+
+  const onClickBlock = ({ value: type }) => {
```

Comment:
For me a couple of comments explaining the semantics would have helped. Otherwise I just assume this is some magic handler :)